cli/lsp/tsc.rs

```diff
@@ -411,6 +413,77 @@ impl QuickInfo {
   }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct RenameLocation {
+  // inherit from DocumentSpan
+  text_span: TextSpan,
+  file_name: String,
+  original_text_span: Option<TextSpan>,
+  original_file_name: Option<String>,
+  context_span: Option<TextSpan>,
+  original_context_span: Option<TextSpan>,
+  // RenameLocation props
+  prefix_text: Option<String>,
+  suffix_text: Option<String>,
+}
+
+pub struct RenameLocations {
+  pub locations: Vec<RenameLocation>,
+}
+
+impl RenameLocations {
+  pub async fn into_workspace_edit(
+    self,
+    ls: &LanguageServer,
```

Comment:
The `RenameLocations` can have the locations that is not same to current document so we should accept `LanguageServer` to resolve `line_index` per each locations.

Reply:
I think you should be able to use the same approach as `DefinitionInfoAndBoundSpan::to_definition`. The `index_provider` argument there is probably what you want.

Reply:
Thank you! I'll fix it.

Reply:
I fixed it with referencing `DefinitionInfoAndBoundSpan::to_definition`.